tests/prepare.sh

```diff
@@ -1,41 +1,85 @@
-# Prepares our environment to run our tests
+#!/bin/bash
+# Prepares a temporary environment to run our tests
 #
-# This script shold be executed once, before running our functinal test
-# for the first time. It'll download and build all needed dependencies
-# to make sure we are not missing anything during our tests.
+# This script should be executed once, before running our functinal test
+# for the first time we are testing a specific commit or in a session, since all the created files are temporary.
+#
+# It'll download and build only utreexod and florestad, since we are not testing the other binaries.
+#
+# Make sure to have python(for the tests), golang(for building utreexod) and rust(for building florestad) installed.
+#
+
+
+# We expect the current dir is the root dir of the project.
+FLORESTA_PROJ_DIR=$(pwd)
+# This helps us to keep track of the actual version being tested without conflicting with any already installed binaries.
+HEAD_COMMIT_HASH=$(git rev-parse HEAD)
+
+TEMP_DIR="/tmp/floresta-integration-tests.${HEAD_COMMIT_HASH}"
+
 
-# Check for dependencies, we need Golang for Utreexod and Rust for Floresta
 go version &>/dev/null
 
 if [ $? -ne 0 ]
 then
 	echo "You must have golang installed to run those tests!"
+	exit 1
 fi
 
+
 cargo version &>/dev/null
 
 if [ $? -ne 0 ]
 then
-	echo "You must have cargo installed to run those tests!"
+	echo "You must have rust with cargo installed to run those tests!"
+	exit 1
+fi
+
+poetry -V  &>/dev/null
+
+if [ $? -ne 0 ]
+then
+	echo "You must have poetry installed to run those tests!"
+	exit 1
 fi
 
 
-mkdir -p ./bin
 
-cd bin
+ls $TEMP_DIR &>/dev/null
+if [ $? -ne 0 ]
+then
+	echo "The tests dir for the tests does not exist. Creating it..."
+	# Dont use mktemp so we can have deterministic results for each version of floresta.
+	mkdir -p "$TEMP_DIR"
+fi
+
+echo "$TEMP_DIR exists. Delete it with"
```

Comment:
Sorry for holding you up on this. But I was running it and found weird that this line was shown right after `The tests dir for the tests does not exist. Creating it...`. This should appear only if the dir does exist, no?

Reply:
Ok, let's leave it as-is. If someone else complains, we change :rofl: